docs/protocol/encryption-specs.md

```diff
@@ -615,6 +615,18 @@ For example, An original contract with a permissioned getter, such that only whi
 
 After a contract runs on the chain, an attaker can sync up a node up to a specific block in the chain, and then call into the enclave with the same authenticated user inputs that were given to the enclave on-chain, but out-of-order, or omit selected messages. A contract that does not anticipate or protect against this might end up deanonimizing the information provided by users. for example, in a naive voting contract (or other personal data collection algorithm), we can deanonimize a voter by re-running the vote without the target's request, and analyze thedifference in final results.
 
+## More Advanced Tx Replay attacks -- search to decision for Millionaire's problem
+
+This attack provides a specific example of a TX replay attack extracting the full information of a client based on replaying a TX.
+
+Specifically, assume for millionaire's that you have a contract where one person inputs their amount of money, then the other person does, then the contract sends them both a single bit saying who's is more -- this is the simplest implementation for Millionaire's problem solving, and it reveals the minimum amount of information (a single bit). As person 2, binary search the interval of possible money amounts person 1 could have -- say you know person 1 has less than 1 trillion dollars. first, query with 500 billion with your node detached from the wider network, get the single bit out (whether the true value is higher or lower), then repeat
```

Comment:
This can be explained better. How can the 2nd player query the contract log(n) times? Taking the node offline is not enough. Also the numbers (1 trillion etc) are irrelevant.